src/Stratis.Bitcoin/BlockPulling/BlockPuller.cs

```diff
@@ -296,7 +296,7 @@ private long GetTotalSpeedOfAllPeersBytesPerSec()
         {
             lock (this.peerLock)
             {
-                return this.pullerBehaviorsByPeerId.Sum(x => x.Value.SpeedBytesPerSecond);
+                return this.pullerBehaviorsByPeerId.Where(p => p.Value.SpeedBytesPerSecond > 0).Sum(x => x.Value.SpeedBytesPerSecond);
```

Comment:
@noescape00 can you comment on this?

Reply:
this change is redundant. 

SpeedBytesPerSecond can't and shouldn't be negative, only >= 0. 

So there is no difference between summing collection of: {0, 5, 8, 0, 0, 3} and collection of {5, 8, 3} .
Please revert the change.

Reply:
@noescape00 please read the PR description on this.

Reply:
I've read. Looks like it's a bug. 
It should be fixed by eliminating the source of the problem, not the consequence.
Ignoring the negative values is not the fix, it's a workaround. 

I've created an issue for it: https://github.com/stratisproject/StratisBitcoinFullNode/issues/2224